### PR TITLE
Add Lap Checkbox to Driver Controls

### PR DESCRIPTION
**Pull request by Mavestri**

### CODE REVIEW

File: src/BusinessLayer/InternetReporting.cpp [hunk 1]
```diff
@@ -191,6 +191,7 @@ QJsonObject InternetReporting::makeDriverControls()
     driverControls.insert("Horn", dataContainerList[packetNum_]->getDriverControlsData().horn());
     driverControls.insert("Reset", dataContainerList[packetNum_]->getDriverControlsData().reset());
     driverControls.insert("Aux", dataContainerList[packetNum_]->getDriverControlsData().aux());
+    driverControls.insert("LapButton", dataContainerList[packetNum_]->getDriverControlsData().lapButton());
```
Comment (bill-luu) [inline src/BusinessLayer/InternetReporting.cpp]: Lets use `Lap` instead of `LapButton`

Comment (mglyngberg) [reply src/BusinessLayer/InternetReporting.cpp]: I am very sorry @Mavestri

Comment (Mavestri) [reply src/BusinessLayer/InternetReporting.cpp]: 😢

Comment (ThomasVy) [reply src/BusinessLayer/InternetReporting.cpp]: Bill, why did you have to make Liam cry?

Comment (bill-luu) [reply src/BusinessLayer/InternetReporting.cpp]: <3

File: .gitignore [hunk 2]
```diff
@@ -7,3 +7,4 @@ release
 *.save
 build*
 *.autosave
+**/.qmake.stash
```
Comment (bill-luu) [inline .gitignore]: Why not just `.qmake.stash`

File: src/Tests/BusinessLayer/SerialReportingTest.cpp [hunk 3]
```diff
@@ -349,7 +349,8 @@ TEST_F(SerialReportingTest, sendDriverControlsTest) // TODO create function whic
                             driverControlsData_->pushToTalk(),
                             driverControlsData_->horn(),
                             driverControlsData_->reset(),
-                            driverControlsData_->aux()
+                            driverControlsData_->aux(),
+                            driverControlsData_->lap()
                           };
     Util::writeBoolsIntoArray(data, 8, driverInputs, 7);
```
Comment (bill-luu) [inline src/Tests/BusinessLayer/SerialReportingTest.cpp]: Shouldn't this also be increased to 8, like in the other file?
` writeBoolsIntoArray(packetPayload, 8, controlsArray, 8);`

File: .gitignore (hunk 2 above)
Comment (Mavestri) [reply .gitignore]: > Why not just `.qmake.stash`

It was in the all the submodule directories, this should just remove `.qmake.stash` from all the submodules.

File: src/Tests/BusinessLayer/SerialReportingTest.cpp (hunk 3 above)
Comment (Mavestri) [reply src/Tests/BusinessLayer/SerialReportingTest.cpp]: > Shouldn't this also be increased to 8, like in the other file?
> ` writeBoolsIntoArray(packetPayload, 8, controlsArray, 8);`

@mglyngberg and I tried setting this to 8 and it stopped working during testing. Tbh not sure why this works, but it does

Comment (mglyngberg) [reply src/Tests/BusinessLayer/SerialReportingTest.cpp]: Even though the test does not verify the contents of the message it should have the same logic as the actual function.

**Pull request merged**